From f7e53998b956c99e875637740002bca25473a6a8 Mon Sep 17 00:00:00 2001
From: "Luis Gustavo S. Barreto" <gustavo@ossystems.com.br>
Date: Tue, 31 Jul 2018 13:44:26 -0300
Subject: [PATCH] updatehub: lock state change mutex in cancel operation

Signed-off-by: Luis Gustavo S. Barreto <gustavo@ossystems.com.br>
---
 updatehub/updatehub.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/updatehub/updatehub.go b/updatehub/updatehub.go
index 2690dd2..fc1300a 100644
--- a/updatehub/updatehub.go
+++ b/updatehub/updatehub.go
@@ -141,6 +141,9 @@ func NewUpdateHub(
 }
 
 func (uh *UpdateHub) Cancel(nextState State) {
+	uh.stateMutex.Lock()
+	defer uh.stateMutex.Unlock()
+
 	uh.state.Cancel(true, nextState)
 }
 
